apps/protocol-api/src/prisma/resolvers/ActivityTypes.ts

```diff
@@ -0,0 +1,59 @@
+import * as TypeGraphQL from 'type-graphql';
+import { ActivityType } from '../generated/type-graphql';
+import { Context } from './types';
+
+@TypeGraphQL.InputType('GetActivityTypesPerUserAndDAOs')
+export class GetActivityTypesPerUserAndDAOs {
+  @TypeGraphQL.Field(_type => Number)
+  userId: number;
+}
+
+@TypeGraphQL.ArgsType()
+export class GetActivityTypesPerUserAndDAOsArgs {
+  @TypeGraphQL.Field(_type => GetActivityTypesPerUserAndDAOs, {
+    nullable: false,
+  })
+  where: GetActivityTypesPerUserAndDAOs;
+}
+
+@TypeGraphQL.ObjectType()
+export class ActivityTypesByUser {
+  @TypeGraphQL.Field(_type => Number)
+  id: number;
+
+  @TypeGraphQL.Field(_type => Boolean)
+  active: boolean;
+
+  @TypeGraphQL.Field(_type => String)
+  name: string;
+
+  @TypeGraphQL.Field(_type => String)
+  createdAt: string;
+
+  @TypeGraphQL.Field(_type => String)
+  updatedAt: string;
+}
+
+@TypeGraphQL.Resolver(_of => ActivityType)
+export class ActivityTypesCustomResolver {
+  @TypeGraphQL.Query(_returns => [ActivityTypesByUser], {
+    nullable: false,
+  })
+  async getActivityTypesByUser(
+    @TypeGraphQL.Ctx() { prisma }: Context,
+    @TypeGraphQL.Args() args: GetActivityTypesPerUserAndDAOsArgs,
+  ) {
+    const user_id = args.where.userId;
+
```

Comment:
We should check that the user in the context is the same as the user_id and also verify the user is part of the requested guilds.